driver/src/main/java/org/kaazing/k3po/driver/netty/bootstrap/http/HttpClientChannelSink.java

```diff
@@ -371,13 +357,14 @@ else if (httpClientConfig.getMaximumBufferedContentLength() > 0) {
         case CONTENT_BUFFERED: {
             HttpRequest httpBufferedRequest = this.httpBufferedRequest;
             this.httpBufferedRequest = null;
-
-            ChannelBuffer httpBufferedContent = httpBufferedRequest.getContent();
-            int httpReadableBytes = httpBufferedContent.readableBytes();
-            setContentLength(httpBufferedRequest, httpReadableBytes);
-            ChannelFuture future = transport.write(httpBufferedRequest);
-            httpClientChannel.state(CONTENT_COMPLETE);
-            chainWriteCompletes(future, httpFuture, httpReadableBytes);
+            if (httpBufferedRequest != null) {
```

Comment:
Looks like the only way to transition to CONTENT_BUFFERED state is when the writer buffers and sets the httpBufferedRequest.  Is there a race condition that causes httpBufferedRequest to be null?